hudi-common/src/main/java/org/apache/hudi/internal/schema/utils/AvroSchemaEvolutionUtils.java

```diff
@@ -140,10 +140,14 @@ public static Schema reconcileSchema(Schema incomingSchema, Schema oldTableSchem
    * @return schema (based off {@code source} one) that has nullability constraints and datatypes reconciled
    */
   public static Schema reconcileSchemaRequirements(Schema sourceSchema, Schema targetSchema, Map<String, String> opts) {
-    if (sourceSchema.getType() == Schema.Type.NULL || sourceSchema.getFields().isEmpty() || targetSchema.getFields().isEmpty()) {
+    if (targetSchema.getType() == Schema.Type.NULL || targetSchema.getFields().isEmpty()) {
```

Comment:
Are these required for this patch or they can be separated out as a separate fix?

Reply:
already landed as a separate pr: https://github.com/apache/hudi/pull/10288